api/src/pcapi/routes/backoffice_v3/user_generation/forms.py

```diff
@@ -1,9 +1,19 @@
+from pcapi.core.subscription.models import SubscriptionStep
 from pcapi.core.users import constants as users_constants
 from pcapi.routes.backoffice_v3.forms import fields
 from pcapi.routes.backoffice_v3.forms import utils
 
 
 class UserGeneratorForm(utils.PCForm):
     age = fields.PCIntegerField("Age", default=users_constants.ELIGIBILITY_AGE_18)
-    is_email_validated = fields.PCSwitchBooleanField("Email validé", default=True)
+    step = fields.PCSelectField(
+        "Étape de validation",
+        choices=[
+            (step.value, step.value.title().replace("-", " "))
+            for step in SubscriptionStep
+            if step != SubscriptionStep.MAINTENANCE
+        ],
+        default=SubscriptionStep.EMAIL_VALIDATION.value,
+        coerce=lambda value: SubscriptionStep(value) if SubscriptionStep.is_valid(value) else None,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏ça résoud quoi comme problème ce coerce avec le `is_valid` ?

Reply:
J'ai mis cette fonction par sécurité parce qu'on est pas sûr de recevoir un élément de valide de l'enum dans un formulaire. N'importe qui pourrait envoyer une autre valeur en POST. La fonction `coerce` permet de contraindre les valeurs possibles et de caster en instance de la classe enum `SubscriptionStep`.